### PR TITLE
Remove write method from IDocumentStorageService

### LINKED ISSUE #9250: Remove IDocumentStorageService.write
This is no longer used and causes another reference to ITree which we want to eliminate. This probably doesn't need to be first deprecated.

### PR DESCRIPTION
**Pull request by kian-thompson**

### CODE REVIEW

**Comment by kian-thompson** [comment]
@vladsud Did I target the correct branch? Also, should I note this breaking change in some `.md` file?

**Comment by vladsud** [comment]
> @vladsud Did I target the correct branch? Also, should I note this breaking change in some .md file?

@heliocliu & @anthony-murphy to confirm RE branch.

yes, it would be great to add a record to /breaking.md. Take a look at its structure (and any previous commit that changed it) - it should be pretty easy to figure out structure. That said, it operates in client version space, and this change is in definitions, I'm not sure which section such update should belong to. Hope Helio & Tony can provide more clarity here.

File: BREAKING.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ ICodeLoader interface was deprecated a while ago and will be removed in the next
 - [Scope is no longer an IFluidObject](#scope-is-no-longer-an-IFluidObject)
 - [IFluidHandle and requestFluidObject generic's default no longer includes IFluidObject](#IFluidHandle-and-requestFluidObject-generics-default-no-longer-includes-IFluidObject)
 - [LazyLoadedDataObjectFactory.create no longer returns an IFluidObject](#LazyLoadedDataObjectFactory.create-no-longer-returns-an-IFluidObject)
-
+- [Remove write method from IDocumentStorageService](#Remove-Write-Method-from-IDocumentStorageService)
```
Comment (heliocliu) [inline BREAKING.md:35]: Following the same format for the 0.59 section, can you create and add this to a new 1.0 section above?  (0.59 release branch already got cut and next is bumping to 1.0 imminently)

**Comment by heliocliu** [comment]
> > @vladsud Did I target the correct branch? Also, should I note this breaking change in some .md file?
> 
> @heliocliu & @anthony-murphy to confirm RE branch.
> 
> yes, it would be great to add a record to /breaking.md. Take a look at its structure (and any previous commit that changed it) - it should be pretty easy to figure out structure. That said, it operates in client version space, and this change is in definitions, I'm not sure which section such update should belong to. Hope Helio & Tony can provide more clarity here.

Currently all breaking changes go into the same breaking section regardless of if they are client or definitions or what.  This might change in the future but makes sense for now because releases are coupled with client versions at the moment.

File: BREAKING.md [hunk 2]
```diff
@@ -129,6 +129,9 @@ LazyLoadedDataObjectFactory.create no longer returns an IFluidObject, it now onl
 
 As a short term fix the return type of this method can be safely casted to an IFluidObject. However, IFluidObject is deprecated and will be removed in an upcoming release so this can only be a temporary workaround before moving to [FluidObject](#Deprecate-IFluidObject-and-introduce-FluidObject).
 
+### Remove Write Method from IDocumentStorageService
+`IDocumentStorageService.write(...)` is no longer used and causes another reference to `ITree` which we want to eliminate.
```
Comment (heliocliu) [inline BREAKING.md:133]: Can you update this to include which package (`driver-definitions`) the affected interface is from, along with notes on what users should do to migrate?  (Even if the migration answer is just to stop using it and there's no alternative, please include that).  If there are preemptive steps users can do to ease the migration, please also make a separate change to add those steps into the upcoming section of the breaking.md file against the `main` branch.  If preemptive migration steps don't make sense here, you can skip that second change

**Review by vladsud — APPROVED** [review]
Looks fine, but you have compat build issue.
I myself struggled with similar issues last time I hit. Please try to follow comment above get_current_InterfaceDeclaration_IDocumentStorageService, but if you hit problems, can you please Helio ask for help?

**Comment by vladsud** [comment]
Glance at https://github.com/microsoft/FluidFramework/pull/9895/files, it has similar in nature change (RE compat impact)

**Review by vladsud — APPROVED** [review]

File: common/lib/driver-definitions/src/test/types/validateDriverDefinitionsPrevious.ts [hunk 3]
```diff
@@ -372,6 +372,7 @@ declare function get_current_InterfaceDeclaration_IDocumentStorageService():
 declare function use_old_InterfaceDeclaration_IDocumentStorageService(
     use: TypeOnly<old.IDocumentStorageService>);
 use_old_InterfaceDeclaration_IDocumentStorageService(
+    // @ts-expect-error compatibility expected to be broken
```
Comment (vladsud) [inline common/lib/driver-definitions/src/test/types/validateDriverDefinitionsPrevious.ts:375]: I'm not an expert here, but I though an appropriate change in package.json should not require this change - please ping Helio for clarification

Comment (heliocliu) [reply common/lib/driver-definitions/src/test/types/validateDriverDefinitionsPrevious.ts:375]: currently the change in the package.json is expected to produce this change as part of the `typetests:gen` script so this is right

**Pull request merged**